LoRaEngine/LoraKeysManagerFacade/CreateEdgeDevice.cs

```diff
@@ -194,22 +76,16 @@ async Task<ConfigurationContent> GetConfigurationContentAsync(Uri configLocation
                 log.LogWarning(ex.Message);
 
                 // In case of an exception in device provisioning we want to make sure that we return a proper template if our devices are successfullycreated
-                var edgeGateway = await this.registryManager.GetDeviceAsync(deviceName);
+                var edgeGateway = await this.registryManager.GetTwinAsync(deviceName);
 
                 if (edgeGateway == null)
                 {
                     return PrepareResponse(HttpStatusCode.Conflict);
                 }
 
-                if (deployEndDevice)
+                if (deployEndDevice && !await this.registryManager.DeployEndDevices())
```

Comment:
I am not sure how idempotent is the operation, anyway you are already doing this operation on line 69. What about setting a local variable defaulting to false to capture the result of the operation?

Reply:
I made the DeployEndDevices idempotent internally.